04.ls/ls.rb

```diff
@@ -51,4 +60,52 @@ def calc_tab_count(entry)
   entry.length / TAB_WIDTH
 end
 
+def print_entries_l(entries)
+  metadatas = entries.map { |entry| extract_entry_metadata(entry) }
+  formated_metadatas = format_metadatas(metadatas, %i[nlink user group size])
+  puts "total #{metadatas.sum { |metadata| metadata[:blocks] }}"
+  formated_metadatas.each do |metadata|
+    puts metadata.values_at(:mode, :nlink, :user, :group, :size, :datetime, :file).join(' ')
+  end
+end
+
+def extract_entry_metadata(entry)
+  stat = File.stat(entry)
+  time = stat.mtime
+  {
+    mode: get_entry_mode(stat),
+    nlink: stat.nlink,
+    user: Etc.getpwuid(stat.uid).name,
+    group: Etc.getgrgid(stat.gid).name,
+    size: stat.size,
+    datetime: time.strftime('%_m %_d %H:%M'),
+    file: entry,
+    blocks: stat.blocks
+  }
+end
+
+def get_entry_mode(stat)
+  mode = format('%06o', stat.mode)
+  file_type = FILE_TYPES[mode[MODE_FILE_TYPE_RANGE]]
+  permission = mode[MODE_PERMISSION_RANGE].chars.map(&:to_i).map do |digit|
+    [
+      (digit & 4).zero? ? '-' : 'r',
+      (digit & 2).zero? ? '-' : 'w',
+      (digit & 1).zero? ? '-' : 'x'
+    ].join
+  end.join
```

Comment:
```suggestion
    ]
  end
```

最後にまとめてjoinするので、ここは配列のまま放置でもいいかもしれません（好みの問題ですが）